### PR TITLE
Set JsonNetVersion for CI build definition

### PR DESCRIPTION
**Pull request by smadala**
## Description
[Build](https://devdiv.visualstudio.com/DevDiv/VS.in%20Agile%20Testing%20IDE/_build/index?buildId=1259374&_a=summary
) failed due to $JsonNetVersion value is empty.

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -530,6 +530,9 @@ function Create-NugetPackages
     # Pass Newtonsoft.Json version to nuget pack to keep the version consistent across all nuget packages.
     $JsonNetVersion = ([xml](Get-Content $env:TP_ROOT_DIR\scripts\build\TestPlatform.Dependencies.props)).Project.PropertyGroup.JsonNetVersion
 
+    # In CI build "Package: TestPlatform SDK" task, "Nuget.exe pack" required JsonNetVersion property for creating nuget package.
+    Write-Host "##vso[task.setvariable variable=JsonNetVersion;]$JsonNetVersion"
```
Comment (codito) [inline scripts/build.ps1]: The knowledge of which CI we're executing in is now coming inside the build script. Is there a way to achieve this without the coupling?

Comment (smadala) [reply scripts/build.ps1]: Added separate script `vsts-prebuild.ps1`.

File: scripts/vsts-prebuild.ps1 [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  $buildSuffix = [string]::Empty
+}
+
+Write-Host "##vso[task.setvariable variable=BuildVersionPrefix;]$buildPrefix"
+Write-Host "##vso[task.setvariable variable=BuildVersionSuffix;]$buildSuffix"
+Write-Host "##vso[task.setvariable variable=PackageVersion;]$packageVersion"
+
+# Set Newtonsoft.Json version to consume in  CI build "Package: TestPlatform SDK" task.
+# "Nuget.exe pack" required JsonNetVersion property for creating nuget package.
+$env:TP_ROOT_DIR = (Get-Item (Split-Path $MyInvocation.MyCommand.Path)).Parent.FullName
```
Comment (singhsarab) [inline scripts/vsts-prebuild.ps1]: We don't need this to an env variable

**Review by singhsarab — APPROVED** [review]
With Suggestion

**Pull request merged**